graphqlapi/src/main/java/io/stargate/graphql/web/resources/GraphqlDmlResource.java

```diff
@@ -143,15 +144,17 @@ private GraphQL getDefaultGraphql(AsyncResponse asyncResponse) {
     }
   }
 
-  private GraphQL getGraphl(String keyspaceName, AsyncResponse asyncResponse) {
+  private GraphQL getGraphl(
+      String keyspaceName, AsyncResponse asyncResponse, HttpServletRequest request) {
     if (!KEYSPACE_NAME_PATTERN.matcher(keyspaceName).matches()) {
       LOG.warn("Invalid keyspace in URI, this could be an XSS attack: {}", keyspaceName);
       // Do not reflect back the value
       replyWithGraphqlError(Status.BAD_REQUEST, "Invalid keyspace name", asyncResponse);
       return null;
     }
 
-    GraphQL graphql = graphqlCache.getDml(keyspaceName);
+    GraphQL graphql =
+        graphqlCache.getDml(keyspaceName, RequestToHeadersMapper.getAllHeaders(request));
```

Comment:
You may need to do a similar in the:`GraphqlDdlResource`
It is using the graphQl cache:
```
this.graphql = graphqlCache.getDdl();
```
and passes it to `GraphqlResourceBase`

Reply:
We don't need to. There are two steps where decorating keyspaces might come into play: building the GraphQL schema, and executing operations.

If we are building the **DML** schema for a keyspace `ks1` that contains a table `foo`, the GraphQL schema will contain a query `foo`, so at construction time we need to inspect the tables. So if keyspace decoration is required, it needs to happen at that time.

On the other hand, the DDL schema only contains generic operations such as `createTable(keyspaceName, tableName)`. Those operations will be the same whichever user keyspaces are present. If you look at `DdlSchemaBuilder`,  it's considerably simpler than `DmlSchemaBuilder`, and at no point does the CQL schema affect the GraphQL schema that we generate.

Execution time is another story: when we process a `createTable(keyspaceName, tableName)`, we need to decorate the keyspace name. This happens in `CassandraFetcher`, we pass the headers when we create the datastore so it will be properly scoped:
```java
    DataStoreOptions dataStoreOptions =
        DataStoreOptions.builder()
            .putAllCustomProperties(httpAwareContext.getAllHeaders())  // <---------------------
            .putAllCustomProperties(authenticationSubject.customProperties())
            .defaultParameters(parameters)
            .alwaysPrepareQueries(true)
            .build();
```